### PR TITLE
Fix behaviour and errors when no product image

### PR DESCRIPTION
**Pull request by t-kelly**
### What are you trying to accomplish with this PR?

Fixes https://github.com/Shopify/slate/issues/99

- Fixes `feature-product.liquid` section by removing placeholder image if no image is set. Also matches `product.liquid` section behaviour by setting the featured image as either the current variant's featured image or the product featured image
- Fixes `product.liquid` by checking if featured image is not black
- Fixes `product.js` by only performing image functions if an image is present on the page. There was some code that was rearranged to make this check easy.

For maintainers:
- [ ] I have :tophat:'d these changes.

### CODE REVIEW

**Comment by t-kelly** [comment]
Added some additional checks:
- Check if product image exists in the featured-collection.liquid
- Check if item image exists in the cart.liquid
- Check if featured image exists in collection.liquid
- Check if featured image exists in search.liquid

File: packages/slate-theme/src/scripts/sections/product.js [hunk 1]
```diff
@@ -29,47 +29,42 @@ theme.Product = (function() {
...
     this.productSingleObject = JSON.parse($(selectors.productJson, this.$container).html());
-    this.settings.imageSize = slate.Image.imageSize($(selectors.productFeaturedImage, this.$container).attr('src'));
 
-    slate.Image.preload(this.productSingleObject.images, this.settings.imageSize);
+    this.$container.on('variantChange' + this.namespace, this.updateAddToCartState.bind(this));
+    this.$container.on('variantPriceChange' + this.namespace, this.updateProductPrices.bind(this));
 
-    this.initVariants();
+    if ($featuredImage.length > 0) {
+      this.settings.imageSize = slate.Image.imageSize($(selectors.productFeaturedImage, this.$container).attr('src'));
```
Comment (chrisberthe) [inline packages/slate-theme/src/scripts/sections/product.js]: The featured product is already cache above:

```js
this.settings.imageSize = slate.Image.imageSize($featuredImage.attr('src'));
```

File: packages/slate-theme/src/scripts/sections/product.js [hunk 2]
```diff
@@ -127,6 +122,7 @@ theme.Product = (function() {
     updateProductImage: function(evt) {
       var variant = evt.variant;
       var sizedImgUrl = slate.Image.getSizedImageUrl(variant.featured_image.src, this.settings.imageSize);
+      var $image = $(selectors.productFeaturedImage, this.$container);
 
       $(selectors.productFeaturedImage, this.$container).attr('src', sizedImgUrl);
```
Comment (chrisberthe) [inline packages/slate-theme/src/scripts/sections/product.js]: Cached on line 125:

```js
$image.attr('src', sizedImgUrl);
```

Either that or omit caching it here altogether.

File: packages/slate-theme/src/scripts/sections/product.js [hunk 3]
```diff
@@ -29,47 +29,42 @@ theme.Product = (function() {
...
-    this.namespace = '.product';
 
     // Stop parsing if we don't have the product json script tag when loading
     // section in the Theme Editor
     if (!$(selectors.productJson, this.$container).html()) {
       return;
     }
 
+    var sectionId = this.$container.attr('data-section-id');
+    var $featuredImage = $(selectors.productFeaturedImage, this.$container);
```
Comment (chrisberthe) [inline packages/slate-theme/src/scripts/sections/product.js]: Maybe we can assign this to `this.$featuredImage` considering we're accessing this element both in the constructor (if it exists) and the `updateProductImage` method below.

File: packages/slate-theme/src/scripts/sections/product.js (hunk 1 above)
Comment (chrisberthe) [inline packages/slate-theme/src/scripts/sections/product.js]: If we go with `this.$featuredImage` above, replace with that.

File: packages/slate-theme/src/scripts/sections/product.js [hunk 4]
```diff
@@ -127,6 +122,7 @@ theme.Product = (function() {
     updateProductImage: function(evt) {
       var variant = evt.variant;
       var sizedImgUrl = slate.Image.getSizedImageUrl(variant.featured_image.src, this.settings.imageSize);
+      var $image = $(selectors.productFeaturedImage, this.$container);
```
Comment (chrisberthe) [inline packages/slate-theme/src/scripts/sections/product.js]: If we go with the instance variable suggestion above, we can omit this altogether.

File: packages/slate-theme/src/templates/cart.liquid [hunk 5]
```diff
@@ -24,10 +24,15 @@
...
           <tr class="responsive-table-row">
+
             <td data-label="{{ 'customer.order.product' | t }}">
-              <a href="{{ item.url | within: collections.all }}">
-                <img src="{{ item | img_url: '240x240' }}" alt="{{ item.title | escape }}">
-              </a>
+
+              {% if item.image != blank %}
+                <a href="{{ item.url | within: collections.all }}">
+                  <img src="{{ item | img_url: '240x240' }}" alt="{{ item.title | escape }}">
```
Comment (chrisberthe) [inline packages/slate-theme/src/templates/cart.liquid]: There's minor inconsistencies in the way we use image tags in Slate. Above, we're using:

```
{{ product.featured_image.src | img_url: '480x480' | img_tag: product.title }}
```

While other instances we're using the HTML `` tag - most likely due to the fact that with the former we're not able to pass a data attribute as a parameter. We could update this to keep it more consistent when not using a data attribute.

File: packages/slate-theme/src/templates/collection.liquid [hunk 6]
```diff
@@ -27,9 +27,11 @@
   <div>
     {% for product in collection.products %}
       <div>
-        <a href="{{ product.url | within: collection }}">
-          <img src="{{ product.featured_image.src | img_url: '480x480' }}" alt="{{ product.featured_image.alt | escape }}">
-        </a>
+        {% if product.featured_image != blank %}
+          <a href="{{ product.url | within: collection }}">
+            <img src="{{ product.featured_image.src | img_url: '480x480' }}" alt="{{ product.featured_image.alt | escape }}">
```
Comment (chrisberthe) [inline packages/slate-theme/src/templates/collection.liquid]: Same here.

**Pull request merged**